doc/source/serve/deployment-graph/visualize_dag_during_development.md

```diff
@@ -0,0 +1,28 @@
+# Pattern: Visualize DAG during development
+
+The example shows how to iteratively develop and visualize your deployment graph.
+
+## Code
+
++++
+
+```{eval-rst}
+.. literalinclude:: ../doc_code/visualize_dag_during_deployment.py
+   :language: python
+```
+
+## Outputs
+
+The node of user choice will become the root of the graph for both execution as well as visualization, where non-reachable nodes from root will be ignored. In the development phase, when we picked `m1_output` as the root, we can see a visualization of the underlying execution path that's partial of the entire graph.
+
+![pic](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/deployment-graph/visualize_partial.svg)
+
+Similarly, when we choose the final dag output, we will capture all nodes used in execution as they're reachable from the root.
+
+![pic](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/deployment-graph/visualize_full.svg)
+
```

Comment:
can you show dagdriver as well?

Reply:
I originally had it there but then removed since it only works for serve but we can re-use all DAG pattern docs for Ray Core level documentation as well, for this reason i removed `serve.start()` from cookbook.

For serve in specific, I will add a dedicated pattern page just for DAGDriver, including default, custom and configuration. What do you think ?